onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -189,18 +193,246 @@ bool LaunchAttentionKernel(
                         batch_size, sequence_length, num_heads, head_size, element_size,
                         reinterpret_cast<const half*>(input), reinterpret_cast<half*>(output), reinterpret_cast<half*>(workspace),
                         mask_index, mask_index_dims, is_unidirectional,
-                        past_sequence_length, reinterpret_cast<const half*>(past), reinterpret_cast<const half*>(extra_add_qk), 
+                        past_sequence_length, reinterpret_cast<const half*>(past), reinterpret_cast<const half*>(extra_add_qk),
                         reinterpret_cast<half*>(present), use_persistent_softmax);
   } else {
     return QkvToContext(prop, cublas, stream,
                         batch_size, sequence_length, num_heads, head_size, element_size,
                         reinterpret_cast<const float*>(input), reinterpret_cast<float*>(output), reinterpret_cast<float*>(workspace),
                         mask_index, mask_index_dims, is_unidirectional,
-                        past_sequence_length, reinterpret_cast<const float*>(past), reinterpret_cast<const float*>(extra_add_qk), 
+                        past_sequence_length, reinterpret_cast<const float*>(past), reinterpret_cast<const float*>(extra_add_qk),
                         reinterpret_cast<float*>(present), use_persistent_softmax);
   }
 }
 
+
+template <typename T>
+bool DecoderQkvToContext(
+  const cudaDeviceProp& prop,
+  cudaStream_t stream,
+  cublasHandle_t& cublas,
+  const size_t element_size,
+  const int batch_size,
+  const int sequence_length,
+  const int kv_sequence_length,
+  const int num_heads,
+  const int head_size,
+  const bool static_kv,
+  const bool use_past,
+  const bool has_layer_state,
+  const bool has_key_padding_mask,
+  const T* gemm_query_buffer,
+  const T* gemm_kv_buffer,
+  const bool* key_padding_mask,
+  const T* key_cache,
+  const T* value_cache,
+  T* qkv_buffer,
+  T* workspace_buffer,
+  T* output,
+  T* new_key_cache,
+  T* new_value_cache)
+{
+  const int max_threads_per_block(prop.maxThreadsPerBlock);
```

Comment:
nit: use the same initializer as others here even though there's no difference for int